firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CommonUtils.java

```diff
@@ -597,6 +603,42 @@ public static String getMappingFileId(Context context) {
     return mappingFileId;
   }
 
+  public static List<BuildIdInfo> getBuildIdInfo(Context context) {
+    // TODO: This functionality should be refactored into an InstanceIdProvider class or similar.
+    List<BuildIdInfo> buildIdInfoList = new ArrayList<>();
+    String[] libNames;
+    String[] arch;
+    String[] buildIds;
+    int libId =
+        CommonUtils.getResourcesIdentifier(context, BUILD_IDS_LIB_NAMES_RESOURCE_NAME, "array");
+    int archId = CommonUtils.getResourcesIdentifier(context, BUILD_IDS_ARCH_RESOURCE_NAME, "array");
+    int buildId =
+        CommonUtils.getResourcesIdentifier(context, BUILD_IDS_BUILD_ID_RESOURCE_NAME, "array");
+
+    if (libId != 0 && archId != 0 && buildId != 0) {
+      libNames = context.getResources().getStringArray(libId);
+      arch = context.getResources().getStringArray(archId);
+      buildIds = context.getResources().getStringArray(buildId);
+
+      if (libNames.length != buildIds.length || arch.length != buildIds.length) {
```

Comment:
What is the purpose of this check?

Reply:
Due to the limitations of the xml that Android resources must conform to, I have to store the arrays separately whic h in theory, e.g. if some enterprising user edited them, they could end up being different lengths and be incoherent.  This simply checks that the lists are all the same length at least.